yetanotherco/aligned_layer
# feat: batcher client message signing

This PR adds signing functionality for batcher client messages.
A new optional CLI argument is added, `--keystore_path`, to specify the path to the keystore. When provided, a message for the keystore password is prompted to the user before sending proofs.

From the batcher side, the message signature is verified, but nothing more is done for the moment. This logic will be added once we integrate with the payment system.

#### Testing
1. Start anvil, aggregator, operator, batcher.
2. Install aligned client locally with `make install_aligned_compiling`
3. Check that everything works as expected when no keystore is provided. You should see a warning log saying that a random private key was created for signing the message. Run
```
 aligned submit --proving_system GnarkPlonkBls12_381 \
--proof batcher/aligned/test_files/plonk_bls12_381/plonk.proof \
--public_input batcher/aligned/test_files/plonk_bls12_381/plonk_pub_input.pub \
--vk batcher/aligned/test_files/plonk_bls12_381/plonk.vk --repetitions 2
```
4. Check that everything works as expected when the keystore is provided, for example:
```
aligned submit --proving_system GnarkPlonkBls12_381 \
--proof batcher/aligned/test_files/plonk_bls12_381/plonk.proof \
--public_input batcher/aligned/test_files/plonk_bls12_381/plonk_pub_input.pub \
--vk batcher/aligned/test_files/plonk_bls12_381/plonk.vk --repetitions 4 --keystore_path config-files/anvil.ecdsa.key.json
```
